ivy/functional/frontends/numpy/ndarray/ndarray.py

```diff
@@ -288,6 +288,9 @@ def flatten(self, order="C"):
         else:
             return np_frontend.ravel(self, order="C")
 
+    def flat(self):
```

Comment:
As this is supposed to be an instance of `numpy.flatiter` class I think we should first implement this class😊.
But if this approach works for now we can implement it later no problem, simply add a ToDo comment

Reply:
Hey Hirwa-nshuti - that was the first question that I had when I started working on the issue: whether to implement the flatiter class first or use the numpy one;. I have updated my flat function to return `self.ivy_array.flat` instead and the test passes for Numpy and fails for the rest of the backends due to `NotImplementedError`; I think this may mean that implementing an ivy flatiter class is needed for all of the backends to access '`flat' `. Please correct me if I am wrong :)

Reply:
Update: I have tried testing the flat property with both the flatten and ravel methods:

`ivy.utils.assertions.check_equal(x.flat, np.nditer(data[0].ravel()))`
`ivy.utils.assertions.check_equal(x.flat, np.nditer(data[0].flatten()))`

and both gave me the same error that I encountered with my current implementation:
<img width="711" alt="update" src="https://github.com/unifyai/ivy/assets/46932156/a75098dd-7526-4753-b875-b96cf09af78f">

Reply:
Basically this class is needed in the ivy frontend so it will be iterator class of frontend ndarray

Reply:
Thank you, Hirwa. Do I implement the class in the ndarray.py file ?